server/src/main/java/org/elasticsearch/cluster/metadata/NodesShutdownMetadata.java

```diff
@@ -106,10 +85,45 @@ public void writeTo(StreamOutput out) throws IOException {
     /**
      * @return A map of NodeID to shutdown metadata.
      */
-    public Map<String, SingleNodeShutdownMetadata> getAllNodeMetadataMap() {
+    public Map<String, SingleNodeShutdownMetadata> getAll() {
         return nodes;
     }
 
+    /**
+     * @return a set of all node ids that might be restarting or shutting down
+     */
+    public Set<String> getAllNodeIds() {
+        return nodes.keySet();
+    }
+
+    /**
+     * @return a shutdown entry for the node if exists with any shutdown type
+     */
+    @Nullable
+    public SingleNodeShutdownMetadata get(String nodeId) {
+        return nodes.get(nodeId);
+    }
+
+    /**
+     * Returns true if the given node is marked as shutting down with any shutdown type.
+     */
+    public boolean contains(String nodeId) {
```

Comment:
NIT: Maybe retain the naming of `isShuttingDown` here? `contains` doesn't really describe the method well?

Reply:
Typical call would already include shutdown in it, I would like to avoid repeating that: `state.metadata().nodeShutdowns().contains(nodeId)`.
I also wanted not to use `shutting down` here because of it is possible confusion with `RESTARTING` type that could be returned by this call.